### ISSUE TITLE
Google AdMob SDK Integration Memory Leak

### ISSUE THREAD

**Issue by ipena-spigot** [AppLovin/AppLovin-MAX-SDK-Android#744]
### MAX SDK Version

13.0.1

### Device/Platform Info

Pixel 4 XL - Android 13

### Current Behavior

Android Studio Profiler tool detects memory leaks after rendering the following Ad formats when using `Google AdMob SDK 23.5.0`:
* Banner
* Interstitial
* Mrec

This is preventing Activities from getting garbage collected and ultimately causing ANRs and crashes

### Expected Behavior

Resources / references should be released such that no memory leaks occur 

### How to Reproduce

Banner:
* MAX Mediation Debugger
* Completed SDK integrations
* Google AdMob (23.5.0)
* Test Mode
* `Banner`
* Load
* Close Activity
* Check for memory leak with Android Studio profiler***

Interstitial
* MAX Mediation Debugger
* Completed SDK integrations
* Google AdMob (23.5.0)
* Test Mode
* `Interstitial`
* Load
* Show
* Close Activity
* Check for memory leak with Android Studio profiler***

Mrec
* MAX Mediation Debugger
* Completed SDK integrations
* Google AdMob (23.5.0)
* Test Mode
* `Mrec` (in MREC vs Native section)
* `Mrec` (In the section with the 5 Ad formats)
* Load
* Close Activity
* Check for memory leak with Android Studio profiler***

Steps for: Check for memory leak with Android Studio***
* In Android Studio
* Profiler
* Analyze Memory Usage
* Start profiler task
* Click on `Detected 2 Leaks`
* `MaxDebuggerMultiAdActivity`
* In the `Instance` tab: `MaxDebuggerMultiAdActivity`
* `References` tab
* Show nearest GC root only
* Expand non `referent in WeakReference` references

### Reproducible in the demo app?

Yes

### Additional Info

Banner `Google AdMob SDK 23.5.0` Memory Leak:
![Screenshot 2024-11-05 at 2 39 44 PM](https://github.com/user-attachments/assets/3980a74e-822d-4ca3-bdcb-6c2f7c5a891a)

Interstitial `Google AdMob SDK 23.5.0` Memory Leak:
![Screenshot 2024-11-05 at 2 59 50 PM](https://github.com/user-attachments/assets/d9752726-0c40-4252-bb9c-636844f17777)

Mrec `Google AdMob SDK 23.5.0` Memory Leak:
![Screenshot 2024-11-05 at 3 26 06 PM](https://github.com/user-attachments/assets/814b2284-6866-44a5-86fd-da1b8f19f14a)

**Comment by Mecze** [comment]
Could this be causing the increase in ANRs lately?
https://github.com/AppLovin/AppLovin-MAX-SDK-Android/issues/714

**Comment by mattn02** [comment]
Hi @ipena-spigot - I was able to reproduce the memory leak. After analyzing the trace, the leak originates from Google AdMob's SDK. To verify this, I observed the same memory leak on [AdMob's demo app](https://github.com/googleads/googleads-mobile-android-examples/tree/main/java/admob/BannerExample) after leaving the activity after the ad was displayed. 

Since this issue stems from AdMob's SDK rather than ours, it would be best for you to reach out to Google AdMob team for support. Thanks!
